django/db/models/query.py

```diff
@@ -904,6 +904,9 @@ def _filter_or_exclude(self, negate, *args, **kwargs):
             assert self.query.can_filter(), \
                 "Cannot filter a query once a slice has been taken."
 
+            if self.query.combinator in ['union', 'intersection', 'difference']:
```

Comment:
I think we want to fail as soon as `combinator` is truthy

```suggestion
            if self.query.combinator:
```